dsp/transform/hilbert_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"fmt"
+	"math"
+	"slices"
+	"testing"
+)
+
+func TestHilbertAnalytic(t *testing.T) {
+	testCases := []struct {
+		input []float64
+	}{
+		{[]float64{0, 0, 0, 0}},
+		{[]float64{1, 2, 3, 4}},
+		{[]float64{1, 2, 3, 4, 5}},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("tc%d", i), func(t *testing.T) {
```

Comment:
```suggestion
	for _, test := range hilbertAnalyticSignalTests {
		t.Run(test.name, func(t *testing.T) {
```
and `s/tc/test/g` elsewhere.